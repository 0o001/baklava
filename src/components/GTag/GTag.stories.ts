import { storiesOf } from '@storybook/vue';
import { withKnobs, text } from '@storybook/addon-knobs';
import GTag from './GTag.vue';

const stories = storiesOf('GTag', module);
stories.addDecorator(withKnobs);

stories
  .addParameters({ component: GTag })
  .add('base', () => ({
    components: { GTag },
    template: `<GTag variant="base">Base Tag</GTag>`,
  }))
  .add('dark', () => ({
    components: { GTag },
    template: `<GTag variant="dark">Dark Tag</GTag>`,
  }))
  .add('success', () => ({
    components: { GTag },
    template: `<GTag variant="success">Success Tag</GTag>`,
  }))
  .add('warning', () => ({
    components: { GTag },
    template: `<GTag variant="warning">Warning Tag</GTag>`,
  }))
  .add('info', () => ({
    components: { GTag },
    template: `<GTag variant="info">Info Tag</GTag>`,
  }))
  .add('error', () => ({
    components: { GTag },
    template: `<GTag variant="error">Error Tag</GTag>`,
  }))
