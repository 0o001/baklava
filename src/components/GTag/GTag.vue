<template>
  <GBox :class="tagColor">
    <GText
      color="white"
      variant="menu-item"
    >
      <slot />
    </GText>
  </GBox>
</template>

<script lang="ts">

import GBox from '../GBox/GBox.vue';
import GText from '../GText/GText.vue';
import $_ from '../../utils/dashUtil.ts';

export default {
  name: 'GTag',
  components: {
    GBox,
    GText,
  },
  props: {
    variant: {
      type: String,
      default: 'base',
    },
  },
  computed: {
    tagColor () {
      return {
        'g-tag': true,
        [$_(this.variant)]: true,
      };
    },
  },
};
</script>

<style lang="scss" scoped>
.g-tag {
  display:inline-block;
  padding: 3px 10px;
  border-radius: var(--radius-lg);
  line-height: normal;
  background-color: var(--mid-grey-500);

  &.-base {
    background-color: var(--mid-grey-500);
  }

  &.-dark {
    background-color: var(--main-grey-500);
  }

  &.-success {
    background-color: var(--green-500);
  }

  &.-info {
    background-color: var(--blue-500);
  }

  &.-warning {
    background-color: var(--orange-500);
  }

  &.-error {
    background-color: var(--red-500);
  }
}
</style>
