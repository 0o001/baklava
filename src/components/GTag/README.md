# GTag

## Slots

| Name      | Description |
| --------- | ----------- |
| `default` | &nbsp;      |

## Props

| Name      | Type     | Description | Default  |
| --------- | -------- | ----------- | -------- |
| `variant` | `String` |             | `"base"` |

## Computed Properties

| Name       | Type     | Description                 |
| ---------- | -------- | --------------------------- |
| `tagColor` | `object` | **Dependencies:** `variant` |

