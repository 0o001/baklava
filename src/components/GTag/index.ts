import GTag from './GTag.vue';

export default GTag;
