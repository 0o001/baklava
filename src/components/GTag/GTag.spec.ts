import { shallowMount } from '@vue/test-utils';
import GTag from './';

describe('GTag', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallowMount(GTag);
  });

  it('should match snapshot when show is prop is given as false ', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render base variant as default when variant prop does not set', () => {
    expect(wrapper.vm.tagColor).toEqual({ 'g-tag': true, '-base': true });
  });

  it('should render given variant when variant prop set', async () => {
    // given
    wrapper.setProps({
      variant: 'success',
    });

    // when
    await wrapper.vm.$nextTick();

    // then
    expect(wrapper.vm.tagColor).toEqual({ 'g-tag': true, '-success': true });
  });
});
