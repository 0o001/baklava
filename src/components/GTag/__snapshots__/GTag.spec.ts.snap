// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GTag should match snapshot when show is prop is given as false  1`] = `
<gbox-stub
  class="g-tag -base"
>
  <gtext-stub
    color="white"
    variant="menu-item"
  />
</gbox-stub>
`;
